pkg/storage/poststorage/interface.go

```diff
@@ -17,6 +17,26 @@ type Interface interface {
 	//
 	SearchComment([]objectid.ID) ([]*Object, error)
 
+	// SearchCreated returns the post objects within the given pagination range in
+	// reversed order of claim creation time. Given the created posts [A B C D E],
+	// the first page [0 3] returns the last posts [C D E].
+	//
+	//     @inp[0] the start paging pointer defining the beginning of the page
+	//     @inp[1] the stop paging pointer defining the end of the page
+	//     @out[0] the list of post objects within the given pagination range
+	//
+	SearchCreated(int, int) ([]*Object, error)
+
+	// SearchExpiry returns the post objects within the given pagination range in
+	// order of claim expiry time. Given the expiring posts [A B C D E], the first
+	// page [0 3] returns the first posts [A B C].
+	//
+	//     @inp[0] the start paging pointer defining the beginning of the page
+	//     @inp[1] the stop paging pointer defining the end of the page
+	//     @out[0] the list of post objects within the given pagination range
+	//
+	SearchExpiry(int, int) ([]*Object, error)
```

Comment:
The storage interface describes how search is ought to work per resource. The resource here are the post objects. The server handlers use the storage to search for the requested data. The storage implementation talks to Redis directly